staging/src/k8s.io/client-go/rest/url_utils.go

```diff
@@ -33,10 +33,7 @@ func DefaultServerURL(host, apiPath string, groupVersion unversioned.GroupVersio
 	}
 	base := host
 	hostURL, err := url.Parse(base)
-	if err != nil {
-		return nil, "", err
-	}
-	if hostURL.Scheme == "" || hostURL.Host == "" {
+	if err != nil || hostURL.Scheme == "" || hostURL.Host == "" {
```

Comment:
If there is not an error but the host is missing, I'm not sure how pre-pending the scheme is going to help.